SimPEG/EM/Static/DC/ProblemDC.py

```diff
@@ -245,16 +326,68 @@ def setBC(self):
                     alpha_ym, alpha_yp = temp_ym, temp_yp
                     alpha_zm, alpha_zp = temp_zm, temp_zp
 
-                    beta_xm, beta_xp = temp_xm*0, temp_xp*0
-                    beta_ym, beta_yp = temp_ym*0, temp_yp*0
-                    beta_zm, beta_zp = temp_zm*0, temp_zp*0
+                    beta_xm, beta_xp = temp_xm*1, temp_xp*1
+                    beta_ym, beta_yp = temp_ym*1, temp_yp*1
+                    beta_zm, beta_zp = temp_zm*1, temp_zp*1
+
+                    gamma_xm, gamma_xp = temp_xm*0., temp_xp*0.
+                    gamma_ym, gamma_yp = temp_ym*0., temp_yp*0.
+                    gamma_zm, gamma_zp = temp_zm*0., temp_zp*0.
+
+                elif(self.bc_type == 'Mixed'):
+                    # Ztop: Neumann
+                    # Others: Mixed: alpha * phi + d phi dn = 0
+                    # where alpha = 1 / r  * dr/dn
+                    # (Dey and Morrison, 1979)
+
+                    # This assumes that the source is located at
+                    # (x_center, y_center_y, ztop)
+                    # TODO: Implement Zhang et al. (1995)
+
+                    xs = np.median(self.mesh.vectorCCx)
+                    ys = np.median(self.mesh.vectorCCy)
+                    zs = self.mesh.vectorCCz[-1]
+                    rxm = 1./np.sqrt(
+                        (gBFxm[:, 0]-xs)**2 + (gBFxm[:, 1]-ys)**2
+                        + (gBFxm[:, 2]-zs)**2
+                        )
```

Comment:
can you move the bracket so it aligns with the rxm? thanks!

eventually, this code should be abstracted a bit, it looks quite repetative for each calc. eg
```
def r_boundary(x, y, z):
    return 1./np.sqrt(
        (x - xs)**2 + (y - ys)**2 + (z - zs)**2
    )

rxp = r_boundary(gBFxp[:, 0], gBFxp[:, 1], gBFxp[:, 2])
```

Reply:
incorporated.